Lib/test/test_pdb.py

```diff
@@ -3104,6 +3112,23 @@ def test_issue42384_symlink(self):
 
             self.assertEqual(stdout.split('\n')[2].rstrip('\r'), expected)
 
+    def test_safe_path(self):
+        """ With safe_path set, pdb should not mangle sys.path[0]"""
+
+        script = textwrap.dedent("""
+            import sys
+            import random
+            print('sys.path[0] is', sys.path[0])
+        """)
+        commands = 'c\n'
+
+
+        with os_helper.temp_cwd() as cwd:
+            stdout, _ = self.run_pdb_script(script, commands, extra_env={'PYTHONSAFEPATH': '1'})
+
+            unexpected = f'sys.path[0] is {os.path.realpath(cwd)}'
+            self.assertNotIn(unexpected, stdout)
```

Comment:
Does this assert really fail in the unfixed case (when it does mangle `sys.path[0]`)? (Sorry, not set up to try it myself right now – ignore me if I’m telling nonsense.) I would expect the script to error out at `import random` already because it can’t find the standard library (in installations where the standard library is at `sys.path[0]`, such as the Windows embeddable one I was using in the bug report) and never get to the `print()`, so stdout still wouldn’t contain the unexpected output. Or is the test never run in such an installation?

Reply:
If `sys.path[0]` contains the stdlib which causes `import random` to fail, `self.run_pdb_script` will fail because the return code would not be 0. The assertion will not even execute.